src/ansys/speos/core/generic/visualization_methods.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (C) 2021 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Provides the ``VisualData`` class."""
+
+from typing import TYPE_CHECKING, List, Union
+
+from ansys.speos.core.generic.general_methods import (
+    Vector,
+    graphics_required,
+    magnitude_vector,
+    normalize_vector,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    import pyvista as pv
+
+
+@graphics_required
+class VisualCoordinateSystem:
```

Comment:
```suggestion
class _VisualCoordinateSystem:
```
You might want to rename it into `_VisualCoordinateSystem` to "notify" users that this class is internal and shouldn't be used by users.